pcsx2/PINE.cpp

```diff
@@ -406,32 +399,32 @@ PINEServer::IPCBuffer PINEServer::ParseCommand(char* buf, char* ret_buffer, u32
 			}
 			case MsgSaveState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_SaveToSlot(FromArray<u8>(&buf[buf_cnt], 0));
+				VMManager::SaveStateToSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgLoadState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_LoadFromSlot(FromArray<u8>(&buf[buf_cnt], 0), false);
+				VMManager::LoadStateFromSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgTitle:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
-				char* title = new char[GameInfo::gameName.size() + 1];
-				sprintf(title, "%s", GameInfo::gameName.ToUTF8().data());
-				const u32 size = strlen(title) + 1;
-				title[size] = 0x00;
+				const std::string& gameName = VMManager::GetGameName();
+				const u32 size = gameName.size() + 1;
+				char* title = new char[size];
+				snprintf(title, size, "%s", gameName.c_str());
```

Comment:
Since you know the size of both `title` and `gameName`, you could even `memcpy` this (and in all instances below).

Reply:
The whole thing thing is redundant - just copy from `gameName` directly into the output buffer.